disable macro checks not used for scanner

We only build wayland-native for the scanner, so disable the bits we don't
actually need. This avoid build issue on older distro such as Centos 5.x:
| error: 'O_CLOEXEC' undeclared (first use in this function)
| error: sys/timerfd.h: No such file or directory
| error: 'CLOCK_MONOTONIC' undeclared (first use in this function)
| error: 'TFD_CLOEXEC' undeclared (first use in this function)
| error: 'SFD_CLOEXEC' undeclared (first use in this function)

Upstream-Status: Pending

Signed-off-by: Ting Liu <b28495@freescale.com>
---
 configure.ac | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -41,16 +41,16 @@ AC_SUBST(GCC_CFLAGS)
 
 AC_CHECK_FUNCS([accept4 mkostemp posix_fallocate])
 
-AC_CHECK_DECL(SFD_CLOEXEC,[],
-	      [AC_MSG_ERROR("SFD_CLOEXEC is needed to compile wayland")],
-	      [[#include <sys/signalfd.h>]])
-AC_CHECK_DECL(TFD_CLOEXEC,[],
-	      [AC_MSG_ERROR("TFD_CLOEXEC is needed to compile wayland")],
-	      [[#include <sys/timerfd.h>]])
-AC_CHECK_DECL(CLOCK_MONOTONIC,[],
-	      [AC_MSG_ERROR("CLOCK_MONOTONIC is needed to compile wayland")],
-	      [[#include <time.h>]])
-AC_CHECK_HEADERS([execinfo.h])
+##AC_CHECK_DECL(SFD_CLOEXEC,[],
+#	      [AC_MSG_ERROR("SFD_CLOEXEC is needed to compile wayland")],
+#	      [[#include <sys/signalfd.h>]])
+#AC_CHECK_DECL(TFD_CLOEXEC,[],
+#	      [AC_MSG_ERROR("TFD_CLOEXEC is needed to compile wayland")],
+#	      [[#include <sys/timerfd.h>]])
+#AC_CHECK_DECL(CLOCK_MONOTONIC,[],
+#	      [AC_MSG_ERROR("CLOCK_MONOTONIC is needed to compile wayland")],
+#	      [[#include <time.h>]])
+#AC_CHECK_HEADERS([execinfo.h])
 
 AC_ARG_ENABLE([scanner],
               [AC_HELP_STRING([--disable-scanner],
-- 
1.8.3.2

